src/sync/impl/sync_metadata.cpp

```diff
@@ -230,4 +264,81 @@ void SyncUserMetadata::remove()
     m_realm = nullptr;
 }
 
+
+SyncFileActionMetadata::SyncFileActionMetadata(const SyncMetadataManager& manager,
+                                               Action action,
+                                               std::string original_name,
```

Comment:
I don't think we should take these strings by value since we're not consuming them. Taking const references instead can probably avoid a copy.